### PR TITLE
Hotfix is not already attending event

### PR DESCRIPTION
**Pull request by christopher-kundrat**
This change is [Reviewable](https://reviewable.io/reviews/ten24/slatwall/5479)

### CODE REVIEW

File: model/process/Order_AddOrderItem.cfc [hunk 1]
```diff
@@ -117,6 +117,21 @@ component output="false" accessors="true" extends="HibachiProcess" {
 
 	// ======================== START: Defaults ============================
 
+	public boolean function isNotAlreadyAttendingEvent(){
+		return !isAlreadyAttendingEvent();
+	}
+
+	public boolean function isAlreadyAttendingEvent(){
+		var account = getOrder().getAccount();
```
Comment (rmarchandTen24) [inline model/process/Order_AddOrderItem.cfc]: if user is a guest then the account will be new.
only run this if account !isNull

File: meta/tests/unit/process/Order_AddOrderItemTest.cfc [hunk 2]
```diff
@@ -48,11 +48,95 @@ Notes:
...
+		};
+		var sku = createPersistedTestEntity("Sku", skuData);
+		orderItem.setSku(sku);
+
+		var eventRegistrationData = {
+			eventRegistrationID = "",
+			account={
+				accountID=account.getAccountID()
+			},
+			eventRegistrationStatusType={
```
Comment (rmarchandTen24) [inline meta/tests/unit/process/Order_AddOrderItemTest.cfc]: add comment to identifiy the type

File: meta/tests/unit/process/Order_AddOrderItemTest.cfc [hunk 3]
```diff
@@ -48,11 +48,95 @@ Notes:
...
+		var eventRegistrationCollectionlist = request.slatwallScope.getService('EventRegistrationService').getEventRegistrationCollectionlist();
+		eventRegistrationCollectionlist.addFilter(orderItemData.orderItemID, accountData.accountID);
+		eventRegistrationCollectionlist.addFilter(orderItemData.orderItemID, skuData.skuID);
+
+		var eventRegistrationData2 = {
+			eventRegistrationID = ""
+		};
+		var eventRegistration2 = createPersistedTestEntity("EventRegistration", eventRegistrationData2);
+		eventRegistration2.setSku(sku);
+
```
Comment (rmarchandTen24) [inline meta/tests/unit/process/Order_AddOrderItemTest.cfc]: remove unused code

Comment (Pete-Eichman) [inline meta/tests/unit/process/Order_AddOrderItemTest.cfc]: removed redundant code

File: meta/tests/unit/process/Order_AddOrderItemTest.cfc (hunk 2 above)
Comment (Pete-Eichman) [inline meta/tests/unit/process/Order_AddOrderItemTest.cfc]: added comment to identify the erst type

File: model/process/Order_AddOrderItem.cfc (hunk 1 above)
Comment (Pete-Eichman) [inline model/process/Order_AddOrderItem.cfc]: changed function to a for loop, which iterates through the registrants, breaking out of the loop by returning true, if the record count is ever not null. loop will keep iterating through the array if still null.

File: model/process/Order_AddOrderItem.cfc [hunk 4]
```diff
@@ -117,6 +117,25 @@ component output="false" accessors="true" extends="HibachiProcess" {
 
 	// ======================== START: Defaults ============================
 
+	public boolean function isNotAlreadyAttendingEvent(){
+		return !isAlreadyAttendingEvent();
+	}
+
+	public boolean function isAlreadyAttendingEvent(){
+		for (registrant in getRegistrants()) {
+			var accountID = registrant.accountID;
```
Comment (christopher-kundrat) [inline model/process/Order_AddOrderItem.cfc]: @Pete-Eichman you need to add a conditional check to make sure that this registrant has an accountID.

File: model/process/Order_AddOrderItem.cfc [hunk 5]
```diff
@@ -117,6 +117,25 @@ component output="false" accessors="true" extends="HibachiProcess" {
...
+
+			var eventRegistrationCollectionlist = getService('eventRegistrationService').getEventRegistrationCollectionlist();
+			eventRegistrationCollectionlist.addFilter('account.accountID',accountID);
+			eventRegistrationCollectionlist.addFilter('sku.skuID',getSku().getSkuID());
+			eventRegistrationCollectionlist.addFilter('eventRegistrationStatusType.systemCode','erstCancelled','!=');
+
+			if(!isNull(eventRegistrationCollectionlist.getRecordsCount())){
+				return true;
+			}
+		}
```
Comment (christopher-kundrat) [inline model/process/Order_AddOrderItem.cfc]: @Pete-Eichman - you'll also want to make sure that this function returns false if it doesn't find an existing registration.

**Pull request closed without merging**